x-pack/legacy/plugins/triggers_actions_ui/np_ready/public/application/lib/capabilities.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/**
+ * NOTE: Applications that want to show the alerting UIs will need to add
+ * check against their features here until we have a better solution. This
+ * will possibly go away with https://github.com/elastic/kibana/issues/52300.
+ */
+
+export function hasShowAlertsCapability(capabilities: any): boolean {
+  if (capabilities.siem && capabilities.siem['alerting:show']) {
```

Comment:
It seems weird that we have `siem` specific conmfiguration specified in the genric Alerting UI.
Is there a wway we can make this configured by the plugins themselves that wish to use the Alerting UI?
Otherwise we're baking in coupling between Alerting and every plugin that uses it, that will make changes harder and more epxensive over time.

Reply:
This was previously discussed with the security team and we decided to avoid adding too much debt until the issue is properly fixed in #43994 via #52300 hopefully in 7.7.